### PR TITLE
Phase 2: TensorGraph add for torch_nntile device

### PR DESCRIPTION
**Pull request by Muxas**
## Summary

Implements **phase 2** of the PyTorch `nntile` device plan: element-wise `add` on `device="nntile"` runs through **libnntile TensorGraph** (`tensor::add` → `TileGraph` → `Runtime`). Phase 1 (PrivateUse1 stub, CPU copy, allocation) is unchanged.

- **`a + b` / `torch.add(a, b, alpha=…)`** on nntile tensors execute via `nntile::tensor::add` (not `NNGraph` autograd)
- PyTorch **C-order** shapes are converted to TensorGraph **storage** layout via `graph_shape_to_storage`
- Gradients remain **PyTorch autograd** (no NNGraph backward in this phase)
- **Optional libnntile link**: set `NNTILE_BUILD_DIR` when building; stub-only install still works for phase 1

## Changes

| Area | Details |
|------|---------|
| `nntile_context.cpp` | Lazy `nntile::Context` init (1 CPU worker) |
| `nntile_executor.cpp` | Eager per-op TensorGraph build + execute for `add` |
| `nntile_add.cpp` | `aten::add.Tensor` / `aten::add.out` for PrivateUse1 |
| `setup.py` | Links `libnntile` + StarPU when `NNTILE_BUILD_DIR` is set |
| Tests | `test_add_parity.py` — CPU vs nntile parity (skipped without libnntile) |
| CI | New `torch-nntile-tensograph-add` job; stub job runs phase-1 tests only |

## Build & test

```bash
# Build libnntile (CPU)
cmake -S . -B build -DUSE_CUDA=OFF -DBUILD_PYTHON_WRAPPERS=OFF ...
cmake --build build --target nntile -j$(nproc)

# Install torch_nntile with libnntile
export NNTILE_BUILD_DIR=$PWD/build NNTILE_SOURCE_DIR=$PWD
export LD_LIBRARY_PATH=$PWD/build/nntile:/opt/starpu/lib
CXX=g++ pip install -e ./torch_nntile --no-build-isolation --force-reinstall

# Run from torch_nntile/ (not repo root — avoids namespace import shadow)
cd torch_nntile && pytest -vv tests
```

## Example

```python
import torch
import torch_nntile

a = torch.tensor([1.0, 2.0], device="nntile")
b = torch.tensor([3.0, 4.0], device="nntile")
z = a + b
assert torch.allclose(z.cpu(), torch.tensor([4.0, 6.0]))
```

## Follow-ups (not in this PR)

- Persistent TensorGraph across ops (vs eager per-op graph today)
- More ATen ops beyond `add`
- PyTorch autograd backward kernels on nntile
- `src/` layout fix for import-from-repo-root footgun

<div><a href="https://cursor.com/agents/bc-523cc3d4-1e73-4e17-b454-8f03109d685f"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-523cc3d4-1e73-4e17-b454-8f03109d685f"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by Muxas** [comment]
cursor created another noise PR
